### PR TITLE
Add Attributes.withAttributes()

### PR DESCRIPTION
**Pull request by jodastephen**
Provide ability to bulk update attributes
Make `SimpleAttributes` public

### CODE REVIEW

File: modules/product/src/main/java/com/opengamma/strata/product/Attributes.java [hunk 1]
```diff
@@ -113,4 +112,21 @@ public static Attributes empty() {
    */
   public abstract <T> Attributes withAttribute(AttributeType<T> type, T value);
 
+  /**
+   * Returns a copy of this instance with the attributes added.
+   * <p>
+   * This returns a new instance with the specified attribute added.
```
Comment (akshaishah) [inline modules/product/src/main/java/com/opengamma/strata/product/Attributes.java]: `This returns a new instance with the specified attributes added.`

File: modules/product/src/main/java/com/opengamma/strata/product/Attributes.java [hunk 2]
```diff
@@ -113,4 +112,21 @@ public static Attributes empty() {
...
+   * The attributes are added using {@code Map.putAll(type, value)} semantics.
+   * 
+   * @param other  the other instance to copy from
```
Comment (akshaishah) [inline modules/product/src/main/java/com/opengamma/strata/product/Attributes.java]: this line has 2 spaces, whereas the line below has 1 space

Comment (jodastephen) [reply modules/product/src/main/java/com/opengamma/strata/product/Attributes.java]: The 2 space rule only applies to `@param`

Comment (akshaishah) [reply modules/product/src/main/java/com/opengamma/strata/product/Attributes.java]: Ah :) didn't realise, apologies!

**Pull request merged**